uniffi_bindgen/src/bindings/gecko_js/mod.rs

```diff
@@ -75,15 +75,38 @@ pub fn generate_bindings(config: &Config, ci: &ComponentInterface) -> Result<Vec
 
     let context = gen_gecko_js::Context::new(config, ci);
 
-    let webidl = WebIdl::new(config, ci)
-        .render()
-        .context("Failed to render WebIDL bindings")?;
+    let dictionaries: Vec<_> = ci
+        .iter_record_definitions()
+        .into_iter()
+        .map(webidl::WebIDLDictionary::new)
+        .collect();
+    let enums = ci.iter_enum_definitions();
+    let functions: Vec<_> = ci
+        .iter_function_definitions()
+        .into_iter()
+        .map(webidl::WebIDLFunction::new)
+        .collect();
+    let interfaces: Vec<_> = ci
+        .iter_object_definitions()
+        .into_iter()
+        .map(webidl::WebIDLInterface::new)
+        .collect();
```

Comment:
You could also consider grouping these together into some sort of `GeckoComponentInterface` struct, by analogy to our `ComponentInterface` struct. Might make the mental model coalesce a little more completely, and make them easier to pass around. It's fine as-is though, if that doesn't sound worthwhile to you.